engine/src/main/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProvider.java

```diff
@@ -82,10 +83,25 @@ public void initCounters() {
     @CacheResult(cacheName = "rbac-recipient-users-provider-get-users")
     public List<User> getUsers(String accountId, boolean adminsOnly) {
         Timer.Sample getUsersTotalTimer = Timer.start(meterRegistry);
+
         List<User> users;
         if (retrieveUsersFromIt) {
-            final List<ITUserResponse> itUserResponses = itUserService.getUsers(accountId, adminsOnly);
-            users = transformToUser(accountId, itUserResponses, adminsOnly);
+            List<ITUserResponse> usersPaging;
+            List<ITUserResponse> usersTotal = new LinkedList<>();
+
+            int pagingStart = 0;
+            int pagingEnd = 10000;
+            do {
+                usersPaging = itUserService.getUsers(accountId, adminsOnly, pagingStart, pagingEnd);
+                usersTotal.addAll(usersPaging);
+
+                pagingStart = pagingEnd + 1;
```

Comment:
This will skip one record if the `firstResult` base index is `0`. For example, after the first iteration, index `10000` will never be retrieved.

Reply:
The variables names are misleading, that's why I made [that comment](https://github.com/RedHatInsights/notifications-backend/pull/1072#discussion_r829853893) which hasn't been applied in all methods BTW.

`pagingEnd` is not a paging end, it's a max results value. If you start at index 0 and ask for 10 results, the IT service will return indexes from `0` to `9`, not from `0` to `10`.